uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -606,6 +607,91 @@ public function testCanSaveALeavePeriodEntitlementWithAComment() {
     $this->unregisterCurrentLoggedInContactFromSession();
   }
 
+  public function testSaveFromCalculationLogsChangesForAnAlreadyExistingPeriodEntitlement() {
+    $userId = 3;
+    $this->registerCurrentLoggedInContactInSession($userId);
+    $type = new AbsenceType();
+    $type->id = 1;
+    $period = new AbsencePeriod();
+    $period->id = 1;
+    $contact = ['id' => 2];
+
+    $periodEntitlement1 = LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contact['id'],
+      'period_id' => $period->id,
+      'type_id' => $type->id,
+      'comment' => 'This is a sample comment'
+    ]);
+
+    $entitlementBalance = 3;
+    $this->createLeaveBalanceChange($periodEntitlement1->id, $entitlementBalance);
+
+    $broughtForward = 1;
+    $proRata = 10;
+    $calculation = $this->getEntitlementCalculationMock(
+      $period,
+      $contact,
+      $type,
+      $broughtForward,
+      $proRata
+    );
+
+    $comment = 'Lorem ipsum dolor sit amet...';
+    LeavePeriodEntitlement::saveFromCalculation($calculation, null, $comment);
+
+    //This should create a record in the entitlement log table
```

Comment:
I think this comment should actually be placed before the `saveFromCalculation()` call, shouldn't it?